runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoPlugin.java

```diff
@@ -349,207 +495,651 @@ private void onChatMessage(ChatMessage event)
 		}
 	}
 
-	private void onGameTick(GameTick event)
+	private boolean isInInferno()
 	{
-		if (client.getMapRegions()[0] != 9043)
-		{
-			return;
-		}
-
-		clearMapAndPriority();
+		return ArrayUtils.contains(client.getMapRegions(), INFERNO_REGION);
+	}
 
-		lastTick = System.currentTimeMillis();
+	int getNextWaveNumber()
+	{
+		return currentWaveNumber == -1 || currentWaveNumber == 69 ? -1 : currentWaveNumber + 1;
+	}
 
-		for (InfernoJad jad : jads)
+	private void calculateUpcomingAttacks()
+	{
+		for (InfernoNPC infernoNPC : infernoNpcs)
 		{
-			jad.gameTick();
-		}
+			infernoNPC.gameTick(client, lastLocation, finalPhase);
 
-		activeHealers.clear();
-		for (NPC npc : client.getNpcs())
-		{
-			if (npc.getId() != NpcID.YTHURKOT_7701 || npc.getInteracting() == client.getLocalPlayer())
+			if (infernoNPC.getType() == InfernoNPC.Type.ZUK && zukShieldCornerTicks == -1)
 			{
-				continue;
+				infernoNPC.updateNextAttack(InfernoNPC.Attack.UNKNOWN, 12); // TODO: Could be 10 or 11. Test!
```

Comment:
Is this still unknown?

Reply:
Yes, I'm waiting for recordings of a Zuk fight to ensure that this is correct.